website/project/views/drafts.py

```diff
@@ -31,6 +32,49 @@
 get_schema_or_fail = lambda query: get_or_http_error(MetaSchema, query)
 autoload_draft = functools.partial(autoload, DraftRegistration, 'draft_id', 'draft')
 
+def validate_embargo_end_date(end_date_string, node):
+    """
+    Our reviewers have a window of time in which to review a draft reg. submission.
+    If an embargo end_date that is within that window is at risk of causing
+    validation errors down the line if the draft is approved and registered.
+
+    The draft registration approval window is always greater than the time span
+    for disallowed embargo end dates.
+
+    :raises: HTTPError if end_date is less than the approval window or greater than the
+    max embargo end date
+    """
+    end_date = parse_date(end_date_string, ignoretz=True)
+    today = datetime.datetime.utcnow()
+    if (end_date - today) <= settings.DRAFT_REGISTRATION_APPROVAL_PERIOD:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid embargo end date',
+            'message_long': 'Embargo end date for this submission must be at least {0} days in the future.'.format(settings.DRAFT_REGISTRATION_APPROVAL_PERIOD)
+        })
+    elif not node._is_embargo_date_valid(end_date):
+        max_end_date = today + settings.DRAFT_REGISTRATION_APPROVAL_PERIOD
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid embargo end date',
+            'message_long': 'Embargo end date must on or before {0}.'.format(max_end_date.isoformat())
+        })
+
+def check_draft_state(draft):
+    if draft.registered_node:
+        raise HTTPError(http.FORBIDDEN, data={
```

Comment:
Is 403 the best return code? This seems like a bad/nonsense request (400).

This isn't a permissions error, insofar as there's no combination of permissions under which the user should ever actually be allowed to access the resource in this case.